pkg/workload/workload.go

```diff
@@ -261,15 +262,26 @@ func dropExcludedResources(input corev1.ResourceList, excludedPrefixes []string)
 	return res
 }
 
+// IsUsingTAS returns information if the workload is using TAS
+func (i *Info) IsUsingTAS() bool {
+	return slices.ContainsFunc(i.TotalRequests,
+		func(ps PodSetResources) bool {
+			return ps.TopologyRequest != nil
+		})
+}
+
 // TASUsage returns topology usage requested by the Workload
-func (i *Info) TASUsage() []TopologyDomainRequests {
-	if !features.Enabled(features.TopologyAwareScheduling) {
-		return nil
-	}
-	result := make([]TopologyDomainRequests, 0)
+func (i *Info) TASUsage() map[kueue.ResourceFlavorReference][]TopologyDomainRequests {
+	result := make(map[kueue.ResourceFlavorReference][]TopologyDomainRequests, 0)
 	for _, ps := range i.TotalRequests {
 		if ps.TopologyRequest != nil {
-			result = append(result, ps.TopologyRequest.DomainRequests...)
+			psFlavors := sets.New[kueue.ResourceFlavorReference]()
+			for _, psFlavor := range ps.Flavors {
+				psFlavors.Insert(psFlavor)
+			}
+			for psFlavor := range psFlavors {
+				result[psFlavor] = append(result[psFlavor], ps.TopologyRequest.DomainRequests...)
+			}
```

Comment:
Hi @mimowo thanks a lot for the quick fix. I'm trying to learn about TAS from your code and I have one question from a beginner's perspective (Maybe I'm asking some stupid question 😅 ):

If we append the same `DomainRequest` for all the flavors in a podset, could that mean we're multiplying the requests by the number of flavors, which can also cause discrepancy compared to the actual request? And if there are multiple flavors, how to we know which of the `DomainRequest` is taken from a specific flavor?

Reply:
Actually, currently we only support one flavor per TopologyRequest for PodSet. This is validated early [here]( https://github.com/kubernetes-sigs/kueue/blob/a672a315eace3eb745848cf2b9cae44156f39664/pkg/scheduler/flavorassigner/tas_flavorassigner.go#L81). This code I have added a bit more generic to accomodate multiple flavors used.

OTOH, it might be misleading, so I'm going to assume only one flavor is returned per topology request also here.

We have it planned to revisit the scenario of multiple flavors, under the [spreadsheet](https://docs.google.com/spreadsheets/d/1MXCjKZtAfqBTb61bJo46u7jIUqRIlu1NrYj1L8Xz-UU/). See "Investigate if possible to have two flavors for the same PodSet".

Reply:
On the second thought I cannot do it, because the TASUsage function returns usage for multiple flavors. Even though there is one flavor per PodSet we still may have multiple flavors used for workloads with multiple PodSets.

Still, I think in the current implementation the usage is accounted properly, the PodSets either:
1. use the same flavor, in that case each case the `DomainRequest` will be appended
2. use different flavors, in that case the usage will be accounted in the corresponding TAS flavors